plugin/pulpcore/plugin/download/http.py

```diff
@@ -1,8 +1,34 @@
+import logging
+
 import aiohttp
+import backoff
 
 from .base import attach_url_to_exception, BaseDownloader, DownloadResult
 
 
+log = logging.getLogger(__name__)
+
+
+logging.getLogger('backoff').addHandler(logging.StreamHandler())
+
+
+def inspect_to_retry(exc):
```

Comment:
I feel like this name is misleading.  The function returns true when a download should **not** be retried, and False if it should.  

A more correct name would be ```is_fatal_exception(exc)``` or something along those lines

Reply:
+1 to changing the name to be more appropriate

Reply:
I agree completely. Let me fix this up.

Reply:
What is the right name for this? Here are some ideas:

* giveup
* maybe_giveup
* inspect_exc_and_giveup
* inspect_exc_and_bail
* inspect_exc_and_fail

Reply:
I'm fine with ```maybe_giveup```

Reply:
'giveup' by itself is the best. @dralley  what do yo uthink?

Reply:
I prefer ```maybe_giveup``` because it's more descriptive.  Giving up isn't a guarantee.  I don't care that much about this one point though.